src/datatype/datatype.cpp

```diff
@@ -112,7 +112,7 @@ size_t Datatype::size() const
   return s;
 }
 
-void Datatype::size(size_t size) const
+void Datatype::set_size(size_t size) const
```

Comment:
Why changing this here to `set_size`? It would be consistent to stay with `size()`.

Reply:
how about "resize" then?
I don't think it's usual to have a setter and getter be called the same thing. And if you think of it as an action, "to size something" doesn't really make sense in English.

Reply:
It is C++ not Java ;)

Reply:
Ok :) Can you point me to where this is established as good practice?
I'll grant that set_ is probably just as obviously bad as get_. But STL uses "resize".

Reply:
I see your point :D As far as I see it the `resize()` method is used to reset the size of a particular container. The `size(size_t )` method, in this case was intended to set the size of a datatype which was never initialized with a particular size. So I feld `resize(size_t)` feels a bit odd to me.

Reply:
Hmm, ok I see. Then yes, they are not semantically equivalent, so maybe we need to think about this some more.
I suppose "reserve" would be just as confusing.
I very likely gravitated to this name because of "H5Tset_size" being as it is. And there might be something to keeping it similar enough to the underlying API.
Also, we could just enforce the size-setting at construction time and not allow the user to resize it afterwards? So String() would produce a variable-size string type definition, whereas String(size_t) would produce a fixed-size one, or something along those lines?

Reply:
That would be an idea. However, in this case I was already thinking of using the `String` class just as a base type and then create two subclasses `VariableString` and `FixedString`. In the first case no size would be required and in the latter case resizing would make no sense.

Reply:
I thought about this also, but I was not sure how the I/O would be handling the polymorphism.

Reply:
That is not a problem. All of theses instances are derived from `Datatype` and thus not a problem.